### ISSUE TITLE
Fix Crystal::HashLiteral#double_splat 

### ISSUE THREAD

**Issue by Blacksmoke16** [crystal-lang/crystal#8266]
Currently `Crystal::HashLiteral#double_splat` maintains the hash `=>` when double splatting.  I think this is isn't really useful, and should work more like `NamedTupleLiteral` where the output would be like `foo: "value", bar: true`.

Or maybe better yet have it as an option similar to the `trailing_comma`, so that the separator value could be changed.

**Comment by jhass** [comment]
Mmh, what's the usecases for this method? Shouldn't we always used named tuples for splatting keyword args?

**Comment by Blacksmoke16** [comment]
@jhass The problem with using name tuples is you can't modify them like you can a hash.

```cr
{% hash = Hash(String, Int32) %}
{% for x, idx in y %}
  {% hash[x] = idx %}
{% end %}
{{hash.double_splat}}
```

Something like that is what I was going for; allows you to build something out at compile time, but have it double splat using the named tuple syntax.

**Comment by jhass** [comment]
Mmh, wasn't there a ticket about being able to modify named tuples at macro time?  What was the outcome there?

**Comment by Blacksmoke16** [comment]
I'm not sure, I don't recall seeing an issue about that?

**Comment by jhass** [comment]
Yeah, I can't find it either, must be recalling a ghost. Anyways, I think I would explore allowing compile time modification of (named) tuples and/or going from hash or array literal to named or regular tuple respectively and remove the splat options from hash and array

**Comment by Blacksmoke16** [comment]
@jhass So I have a working implementation of converting array/hash to tuple/named tuple in macros but I'm kinda conflicted on how to best resolve this issue.

I think having extra methods on macro literal types is a good thing, such as `ArrayLiteral#splat` versus that just being on `TupleLiteral`.  However it does feel a bit strange to have `ArrayLiteral` and `TupleLiteral` be essentially the same thing, while `HashLiteral` and `NamedTupleLiteral` differ in how they get double splatted.  I feel like it would just make more sense to make double splatting a `HashLiteral` work like `NamedTupleLiteral`.
